pkg/webhook/network/network.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network
+
+import (
+	"github.com/gardener/gardener-extensions/pkg/webhook"
+	extensionswebhook "github.com/gardener/gardener-extensions/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+const (
+	// WebhookName is the webhook name.
+	WebhookName = "network"
+	// NameSuffix is a common suffix for all webhook names.
+	NameSuffix = "extensions.gardener.cloud"
+	// ShootProviderLabel is a label on shoot namespaces in the seed cluster that identifies the Shoot cloud provider.
+	// TODO Move this constant to gardener/gardener
+	ShootCloudProviderLabel = "shoot.gardener.cloud/provider"
+	// NetworkProviderLabel is a label on shoot namespaces in the seed cluster that identifies the Shoot network provider.
+	// TODO Move this constant to gardener/gardener
```

Comment:
Can you please create a PR on gardener?

Reply:
Can you vendor the latest version of Gardener and use it?

Reply:
If I do that, wouldn't that override the toml file after the merge to a commit hash?

Reply:
Maybe its better if we wait till there is a release and use a pinned versioned?

Reply:
The next release will take a while, I guess. You're right, yes, generally it's better, but in the current development process with extensibility we often move things around, so I think it's okay to use a commit hash. We've done that multiple times in the past anyways.